backend/graphql/resolvers.ts

```diff
@@ -0,0 +1,166 @@
+        }
+      };
+    },
+    track: async (_parent: any, args: { slug: string }, context: any) => {
+      return getTrack({ ...context.request, params: { slug: args.slug } }, context.reply);
+    },
+  },
+  Mutation: {
+    createTrack: async (_parent: any, args: any) => {
+      const { title, artist, album, genres, coverImage } = args.input;
+      if (!title || !artist || !genres) {
+        throw new Error('Title, artist, and genres are required');
+      }
+      const slug = createSlug(title);
+      const newTrack = await createTrack({
+        title,
+        artist,
+        album,
+        genres,
+        coverImage,
+        slug
+      });
+      return newTrack;
+    },
+    updateTrack: async (_parent: any, args: any) => {
+      const { id, input } = args;
+      const existingTrack = await getTrackById(id);
+      if (!existingTrack) {
+        throw new Error('Track not found');
+      }
+      let updates = { ...input };
+      if (input.title && input.title !== existingTrack.title) {
+        const { createSlug } = await import('../utils/slug');
+        updates.slug = createSlug(input.title);
+      }
+      const updated = await updateTrack(id, updates);
+      if (!updated) {
+        throw new Error('Failed to update track');
+      }
+      return updated;
+    },
+    deleteTrack: async (_parent: any, args: any) => {
+      const { id } = args;
+      const result = await deleteTrack(id);
+      return result;
+    },
+    deleteTracks: async (_parent: any, args: any) => {
+      const { ids } = args;
+      const result = await deleteMultipleTracks(ids);
+      return result;
+    },
+    uploadTrackFile: async (_parent: any, args: any) => {
+         const { id, file } = args;
+
+         try {
+            // Await the file upload
+            const upload = await file;
+            const { filename, mimetype, createReadStream } = upload.file;
+
+            // Validate file type
+            const allowedMimeTypes = [
+               'audio/mpeg',
+               'audio/wav',
+               'audio/mp3',
+               'audio/x-wav',
+            ];
+            console.log(
+               '---------------------------UPLOAD:---------------------------',
+               { upload }
+            );
+            console.log(
+               '---------------------------Uploaded file:---------------------------',
+               { filename, mimetype }
+            );
+
+            if (!allowedMimeTypes.includes(mimetype)) {
+               throw new Error(
+                  'Invalid file type. Only MP3 and WAV files are allowed.'
+               );
+            }
+
+            // Read file stream
+            const stream = createReadStream();
+            const chunks: Buffer[] = [];
+
+            for await (const chunk of stream) {
+               chunks.push(chunk);
+            }
+
+            const buffer = Buffer.concat(chunks);
+            const maxSize = 10 * 1024 * 1024; // 10MB
```

Comment:
This can also be separated in constants file, considering that you don't need to calculate it every time this function is called.